### PR TITLE
Clean up encoding of natural numbers used in array sizes

### PR DESCRIPTION
**Pull request by jonas**
Constant array sizes can only be positive so remove signed conversion
of integer to Scala Native natural number encoding in favour of the
unsigned one.

Refs #39

### CODE REVIEW

File: Utils.h [hunk 1]
```diff
@@ -15,35 +15,17 @@ inline std::string basename(const std::string& pathname) {
...
-    }
-}
-
-inline std::string uint64ToScalaNat(uint64_t v, std::string accumulator = ""){
 
-    if(v > 0){
-        int last_digit = v % 10;
-        int rest = v / 10;
+    int last_digit = v % 10;
+    int rest = v / 10;
```
Comment (kornilova-l) [inline Utils.h]: CLion gives warnings about type casts. Is it fine to change these lines to following: ?
```c++
int last_digit = static_cast<int>(v % 10);
uint64_t rest = v / 10;
```

Comment (jonas) [reply Utils.h]: How about using `auto`? That should keep everything unsigned.

Comment (jonas) [reply Utils.h]: Also, maybe we should configure some compiler warning flags and fail the build.

**Comment by jonas** [comment]
I've enabled `-Wall -Wconversion -Werror` and fixed all warnings.

LLVM headers are marked as system headers to not show warnings for them.

File: ir/Struct.cpp [hunk 2]
```diff
@@ -24,12 +24,10 @@ TypeDef Struct::generateTypeDef() const {
...
-    auto fieldsCount = static_cast<int>(fields.size());
-    for (int i = 0; i < fieldsCount; i++) {
-        s << fields[i].getType();
-        if (i < fieldsCount - 1) {
-            s << ", ";
-        }
+    std::string sep = "";
+    for (const auto &field : fields) {
+        s << sep << field.getType();
+        sep = ", ";
```
Comment (kornilova-l) [inline ir/Struct.cpp]: This is a better way to add commas :thumbsup:

**Pull request merged**